adapter_test.go

```diff
@@ -87,6 +119,33 @@ func TestAdapter_DirectMessages(t *testing.T) {
 	assert.Equal(t, expectedEvt, events[0])
 }
 
+func TestAdapter_IgnoreDirectOwnMessages(t *testing.T) {
```

Comment:
At first it was not clear to me either how the two tests differ until I realized its really only the channel names.
I think the duplication can be removed and the test can be clarified like this:

```go
func TestAdapter_IgnoreOwnMessages(t *testing.T) {
	cases := map[string]struct{ Channel string }{
		"channel message": {"C1H9RESGL"}, // map test case name to channel ID
		"direct message":  {"D023BB3L2"}, // direct slack channels start with a "D"
	}

	for name, c := range cases {
		t.Run(name, func(t *testing.T) {
			brain := joetest.NewBrain(t)
			a, _ := newTestAdapter(t)

			done := make(chan bool)
			go func() {
				a.handleSlackEvents(brain.Brain)
				done <- true
			}()

			evt := &slack.MessageEvent{
				Msg: slack.Msg{
					Text:    "Hello world",
					Channel: c.Channel,
					User:    a.userID,
				},
			}

			a.events <- slack.RTMEvent{Data: evt}

			close(a.events)
			<-done
			brain.Finish()

			assert.Empty(t, brain.RecordedEvents())
		})
	}
}
```

Note that I also used `a.userID` in the event instead of the magic `"42"` because otherwise it requires the reader of the test to know that this ID is actually the test adapters ID which is set in a different function.